docs/pipelines.md

```diff
@@ -316,6 +317,47 @@ tasks:
       name: build-push
 ```
 
+### Guard `Task` execution using `WhenExpressions`
+
+To run a `Task` only when certain conditions are met, it is possible to _guard_ task execution using
+the `when` field. The `when` field allows you to list a series of references to `WhenExpressions`.
+
+The components of `WhenExpressions` are `Input`, `Operator` and `Values`:
+
+- `Input` is the input for the `Guard` checking which can be static inputs or variables (`Parameters` or `Results`).
+- `Values` is an array of string values. The `Values` array must be non-empty. It can contain static values
+or variables (`Parameters` or `Results`).
+- `Operator` represents an `Input`'s relationship to a set of `Values`. `Operators` we will use in `Guards`
+are `In` and `NotIn`.
+
+The declared `WhenExpressions` are evaluated before the `Task` is run. If all the `WhenExpressions`
+evaluate to `True`, the `Task` is run. If any of the `WhenExpressions` evaluate to `False`, the `Task` is
+skipped.
```

Comment:
Since this is part of the `pipeline` docs, we should explain what happens to the pipeline if a task is skipped. Does pipeline continue with the other branches? Does pipeline stop execution entirely? Does this impact the status of Pipeline? Please add such explanation here. 🙏

Reply:
it is currently vague because I'm planning to add the details of what happens in the skipping (continueAfterSkip) pull request -- which should come out shortly after this one -- but will add details here if that's delayed

Reply:
I would still like to see it here independent of `continueAfterSkip`, the idea is to educate users about this feature just by itself (without having them specify `continueAfterSkip`)